### ISSUE TITLE
What is stack build on macos

### ISSUE THREAD

**Issue by bharathyes** [kmonad/kmonad#297]
I am new to macos and have been trying to install kmonad to get Extend layout.

I am following the installation steps and am stuck at the final step:
`stack build --flag kmonad:dext --extra-include-dirs=c_src/mac/Karabiner-DriverKit-VirtualHIDDevice/include/pqrs/karabiner/driverkit:c_src/mac/Karabiner-DriverKit-VirtualHIDDevice/src/Client/vendor/include
`

It just says stack is an unknown command. I have tried searching for `stack build macos` and various iterations and haven't seen any cli tool with that name.

Any help on how to complete this?

**Comment by Vermoot** [comment]
You need to install `stack` using `brew install stack`.

If you don't have homebrew, [you need to install that before then](https://brew.sh/index_fr).

**Comment by pataquets** [comment]
@bharathyes You can also try [building with Docker](https://github.com/kmonad/kmonad/blob/master/doc/installation.md#using-docker) and report back, since nobody has tried it on macOS yet. I don't even know if it would work, but it would be great to know.

You might need to customize the `Dockerfile`, though. I would recommend try building unmodified first to check it works and start tinkering after that.

**Comment by MuhammedZakir** [comment]
> @bharathyes You can also try [building with Docker](https://github.com/kmonad/kmonad/blob/master/doc/installation.md#using-docker) and report back, since nobody has tried it on macOS yet. I don't even know if it would work, but it would be great to know.
> 
> You might need to customize the `Dockerfile`, though. I would recommend try building unmodified first to check it works and start tinkering after that.

I had very bad experiences with Docker in mac. Unless you already use Docker, or planning to use it, you are better of installing Stack. Still, I haven't used it for a few months, and things might have improved.

**Comment by pataquets** [comment]
You're right, @MuhammedZakir . I've never used macOS, but by experiences from colleagues I've heard mixed opinions, some were bad but also others were OK. Although this is by no means statistically relevant, those experiences were *always* when *running* services/apps on Docker, which is relevant here.
Since the build instructions are just for that, ie just building, it's a way easier test, because it will either fail to buld or spit out a working binary, both without leaving any trace on the system. And since the project needs some help in ways to build binaries, the feedback would be very valuable.
Of course, that is true only if @bharathyes has already Docker installed. I'm not expecting him to install just to test it. But since it might help him, it would be a win-win situation.

**Comment by slotThe** [comment]
Seems solved, so I'm closing
